identity/src/main/java/com/stripe/android/identity/ui/DocSelectionScreen.kt

```diff
@@ -0,0 +1,198 @@
+                    remember { requireNotNull(verificationPageState.data).requireSelfie() }
+                Column(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .padding(
+                            start = dimensionResource(id = R.dimen.page_horizontal_margin),
+                            end = dimensionResource(id = R.dimen.page_horizontal_margin),
+                            top = dimensionResource(id = R.dimen.page_vertical_margin),
+                            bottom = dimensionResource(id = R.dimen.page_vertical_margin)
+                        )
+                ) {
+                    Text(
+                        text = documentSelect.title,
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(
+                                top = 58.dp,
+                                bottom = 32.dp
+                            )
+                            .semantics {
+                                testTag = docSelectionTitleTag
+                            },
+                        fontSize = 24.sp,
+                        fontWeight = FontWeight.Bold
+                    )
+                    if (documentSelect.idDocumentTypeAllowlist.count() > 1) {
+                        MultiSelection(
+                            documentSelect.idDocumentTypeAllowlist,
+                            requireSelfie = requireSelfie,
+                            onDocTypeSelected = onDocTypeSelected
+                        )
+                    } else {
+                        SingleSelection(
+                            allowedType = documentSelect.idDocumentTypeAllowlist.entries.first().key,
+                            buttonText = documentSelect.buttonText,
+                            bodyText = documentSelect.body,
+                            requireSelfie = requireSelfie,
+                            onDocTypeSelected = onDocTypeSelected
+                        )
+                    }
+                }
+            }
+            else -> {
+                // Unreachable as DocSelection can only be reached after Consent, at this point
+                // a successful VerificationPage is already saved locally
+            }
+        }
+    }
+}
+
+@Composable
+internal fun MultiSelection(
+    idDocumentTypeAllowlist: Map<String, String>,
+    requireSelfie: Boolean,
+    onDocTypeSelected: (Type, Boolean) -> Unit
+) {
+    var selectedTypeValue by remember { mutableStateOf(SELECTION_NONE) }
+
+    for ((allowedType, allowedTypeValue) in idDocumentTypeAllowlist) {
+        Box(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(
+                    top = 8.dp,
+                    bottom = 8.dp
+                )
+                .semantics {
+                    testTag = allowedType
+                }
+        ) {
+            when (selectedTypeValue) {
```

Comment:
Would it make sense to refactor this to avoid the need for a `when` statement? I find that the composables for each state are similar enough to be rendered like below:

```kotlin
for ((allowedType, allowedTypeValue) in idDocumentTypeAllowlist) {
  val isSelected = allowedType == selectedTypeValue

  // Or maybe a clickable Row instead, as this makes the layout more explicit?
  Box {
    TextButton(
      // …
    ) {
      // …
    }

    if (isSelected) {
      CircularProgressIndicator(
        // …
      )
    }
  }
}
```

Reply:
Done